.github/workflows/buf-ci.yaml

```diff
@@ -0,0 +1,29 @@
+name: buf-ci
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
```

Comment:
unless you are creating a bunch of tags that start with v that aren't version tags, I think you can simplify this to just `v*`.